src/main/java/org/embulk/output/elasticsearch/ElasticsearchHttpClient.java

```diff
@@ -182,11 +189,27 @@ public String getEsVersion(PluginTask task)
         return response.get("version").get("number").asText();
     }
 
+    public int getEsMajorVersion(PluginTask task)
+    {
+        try {
+            if (ES_CURRENT_MAJOR_VERSION > 0) {
+                return ES_CURRENT_MAJOR_VERSION;
+            }
+
+            final String esVersion = getEsVersion(task);
+            ES_CURRENT_MAJOR_VERSION = Integer.parseInt(esVersion.substring(0, 1));
+            return ES_CURRENT_MAJOR_VERSION;
```

Comment:
Better to use one return point. 
```
        if (ES_CURRENT_MAJOR_VERSION <= 0) {
            final String esVersion = getEsVersion(task);
            ES_CURRENT_MAJOR_VERSION = Integer.parseInt(esVersion.substring(0, 1)); 
        }
        
        return ES_CURRENT_MAJOR_VERSION;
```

Reply:
The first return to return the cached value and the second one fetch from API. So I prefer to keep it as is